feat: Pause media upload when internet connection unavailable

dcalhoun opened the pull request:
## Related 

* https://github.com/WordPress/gutenberg/pull/56937
* https://github.com/wordpress-mobile/gutenberg-mobile/pull/6491

## Description

Pause media upload when internet connection is unavailable.

## Testing Instructions
See https://github.com/WordPress/gutenberg/pull/56937.

## Regression Notes
1. Potential unintended areas of impact
    Disrupting successful media uploads, breaking other media block types. 
2. What I did to test those areas of impact (or what existing automated tests I relied on)
    Uploaded media to various block types. 
3. What automated tests I added (or what prevented me from doing so)
    No, opted to test the JavaScript logic instead. 

## PR submission checklist

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] VoiceOver.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

dcalhoun committed:
build: Update Gutenberg ref

dcalhoun committed:
feat: Add paused media upload state

When internet connection is unavailable, pause the upload rather than
marking it as failed.

dcalhoun committed:
feat: Conditionally apply paused media upload state for remote status

Prevent conflicting media upload statuses by applying the paused media
upload logic in additional locations.

dcalhoun committed:
build: Update Gutenberg ref

dcalhoun committed:
docs: Add release note

dcalhoun committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-iOS into feat/update-image-block-upload-visuals

dcalhoun committed:
build: Update Gutenberg ref

derekblank reviewed (approved):
LGTM. 🚀

mokagio committed:
Merge remote trunk into feat/update-image-block-upload-visuals

This is necessary to get up to date with a CI configuration change that
requires running on Xcode 15.1.

See https://github.com/wordpress-mobile/WordPress-iOS/pull/22270

mokagio commented:
Hey @dcalhoun 👋 You might notice a merge `trunk` commit here from me. That is to bring this branch up to date with the latest CI changes on `trunk` and ensure everything works on the new Xcode 15.1 CI stack.

Don't hesitate to ping me if there's any trouble.

derekblank committed:
build: Update Gutenberg ref and merge trunk

dcalhoun committed:
build: Update Gutenberg ref

dcalhoun committed:
fix: Revert seemingly errant update to Podfile.lock causing CI failures

The Podfile.lock changes in the branch were unexpected and causing CI
failures.

This changeset was created by the following:

```
git checkout origin/trunk -- Podfile.lock
bundle install
bundle exec pod install
```
